bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestCallOptionsFactory.java

```diff
@@ -31,81 +33,92 @@
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 
-/** Tests for {@link CallOptionsFactory}. */
+/**
+ * Tests for {@link CallOptionsFactory}.
+ */
 @RunWith(JUnit4.class)
 public class TestCallOptionsFactory {
 
-    @Mock
-    ScheduledExecutorService mockExecutor;
+  private static final long NOW = System.nanoTime();
+
+  @Mock
+  ScheduledExecutorService mockExecutor;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+  }
 
-    @Before
-    public void setup() {
-        MockitoAnnotations.initMocks(this);
-    }
+  @Test
+  public void testDefault() {
+    CallOptionsFactory factory = new CallOptionsFactory.Default();
+    Assert.assertSame(CallOptions.DEFAULT, factory.create(null, null));
+  }
 
-    @Test
-    public void testDefault() {
+  @Test
+  public void testDefaultWithContext() {
+    Deadline deadline = DeadlineUtil.deadlineWithFixedTime(1, TimeUnit.SECONDS, NOW);
+    Context.CancellableContext context = Context.current().withDeadline(deadline, mockExecutor);
+    context.run(new Runnable() {
+      @Override
+      public void run() {
         CallOptionsFactory factory = new CallOptionsFactory.Default();
-        Assert.assertSame(CallOptions.DEFAULT, factory.create(null, null));
-    }
+        assertSameTimeDiff(
+                Context.current().getDeadline().timeRemaining(TimeUnit.MILLISECONDS),
+                getDeadlineMs(factory, null)
+        );
+      }
+    });
+  }
 
-    @Test
-    public void testDefaultWithContext() {
-        Context.CancellableContext context = Context.current().withDeadlineAfter(1, TimeUnit.SECONDS, mockExecutor);
-        context.run(new Runnable() {
-            @Override
-            public void run() {
-                CallOptionsFactory factory = new CallOptionsFactory.Default();
-                Assert.assertEquals(
-                        Context.current().getDeadline().timeRemaining(TimeUnit.MILLISECONDS),
-                        getDeadlineMs(factory, null),
-                        100.0
-                );
-            }
-        });
-    }
+  @Test
+  public void testConfiguredDefaultConfig() {
+    CallOptionsConfig config = new CallOptionsConfig.Builder().build();
+    CallOptionsFactory factory = new CallOptionsFactory.ConfiguredCallOptionsFactory(config);
+    Assert.assertSame(CallOptions.DEFAULT, factory.create(null, null));
+  }
 
-    @Test
-    public void testConfiguredDefaultConfig() {
-        CallOptionsConfig config = new CallOptionsConfig.Builder().build();
-        CallOptionsFactory factory = new CallOptionsFactory.ConfiguredCallOptionsFactory(config);
-        Assert.assertSame(CallOptions.DEFAULT, factory.create(null, null));
-    }
+  @Test
+  public void testConfiguredConfigEnabled() {
+    CallOptionsConfig config = new CallOptionsConfig.Builder().setUseTimeout(true).build();
```

Comment:
Not sure if you have addressed my previous comment.
Set ShortRpcTimeoutMs implicitly? Easier to see where it is coming from.